src/impls.rs

```diff
@@ -240,12 +239,9 @@ impl<T> TypeInfo for PhantomData<T>
 where
     T: TypeInfo + ?Sized + 'static,
 {
-    type Identity = Self;
+    type Identity = T;
```

Comment:
I think this is wrong, should be `Self`. Otherwise `PhantomData<bool>` == `bool`